### PR TITLE
Obmin 352 header redesign

### PR DESCRIPTION
**Pull request by ASGladchenko**

### CODE REVIEW

**Review by Kyslytsia — APPROVED** [review]

File: frontend/src/components/common/search-context/index.tsx [hunk 1]
```diff
@@ -4,7 +4,21 @@
 import { createContext, useState, useMemo, useEffect } from 'react';
 import { useSearchParams } from 'react-router-dom';
 
-const SearchContext = createContext();
+interface IProvider {
+  search: string;
```
Comment (cloudmenson) [inline frontend/src/components/common/search-context/index.tsx:8]: Why didn't you put IProvider in types file?

**Review by cloudmenson — APPROVED** [review]

File: frontend/src/app/index.tsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ import { ModalProvider, SearchProvider } from 'src/components/common';
 
 import Routes from 'src/routes/routes';
 import Footer from 'src/components/footer';
-import Header from 'src/components/header';
+import { HeaderR } from 'src/components/new_header';
```
Comment (Wolshebnik) [inline frontend/src/app/index.tsx:13]: change name

File: frontend/src/components/new_header/config.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { getTranslatedText } from '../local';
+import { route } from './../../routes/routeConstants';
```
Comment (Wolshebnik) [inline frontend/src/components/new_header/config.ts:2]: use absolute path

File: frontend/src/components/new_header/config.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    link: route.home,
+    text: getTranslatedText('header.another'),
+  },
+];
+
+export const burgerLinks = [
+  { to: '/', text: getTranslatedText('header.account') },
+  { to: '/', text: getTranslatedText('button.addAdv'), mobile: true },
+  { to: '/', text: getTranslatedText('header.goodness'), icon: true },
+  { to: '/', text: getTranslatedText('header.about') },
```
Comment (Wolshebnik) [inline frontend/src/components/new_header/config.ts:43]: why is path the same?
and path should be from constant

File: frontend/src/components/new_header/index.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+import { SearchContext } from 'src/components/common';
+import { getUserThunk } from 'src/store/profile/thunk';
+import { useAppDispatch, useAppSelector } from 'src/store';
+import { getAuth, getAuthed, setLanguage } from 'src/store/auth/slice';
+
+import * as Styles from './styles';
+import { getTranslatedText } from '../local';
+import { SelectLanguage } from '../selectLang';
+import { burgerLinks, categories } from './config';
+import { useWindowSize } from 'react-use';
```
Comment (Wolshebnik) [inline frontend/src/components/new_header/index.tsx:26]: sort

File: frontend/src/components/new_header/index.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+const HeaderR = () => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const { pathname } = useLocation();
+  const { width } = useWindowSize();
```
Comment (Wolshebnik) [inline frontend/src/components/new_header/index.tsx:32]: sort

**Review by Kyslytsia — APPROVED** [review]

**Review by Mish-hash — APPROVED** [review]

File: frontend/src/components/header/config.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    link: route.home,
+    text: getTranslatedText('header.another'),
+  },
+];
+
+export const burgerLinks = [
+  { to: '/', text: getTranslatedText('header.account') },
+  { to: '/', text: getTranslatedText('button.addAdv'), mobile: true },
+  { to: '/', text: getTranslatedText('header.goodness'), icon: true },
+  { to: '/', text: getTranslatedText('header.about') },
```
Comment (Wolshebnik) [inline frontend/src/components/header/config.ts:43]: why all references are not from a constant

**Review by Wolshebnik — APPROVED** [review]

**Pull request merged**